### PR TITLE
added search functionality

### PR DESCRIPTION
**Pull request by JantuDeb**
# Added search filter
- search video on the home page.
- fixed bug from the last commit

### CODE REVIEW

File: src/components/navbar/Navbar.jsx [hunk 1]
```diff
@@ -45,7 +70,11 @@ const Navbar = ({ hideHamburgerMenu }) => {
           {authState.isLogedIn ? (
             <Link to="/profile">
               {authState.user.photo ? (
-                <img src={authState.user.photo?.secure_url} alt=""  className="avatar-small radius-full"/>
+                <img
+                  src={authState.user.photo?.secure_url}
+                  alt=""
```
Comment (rkapoor10) [inline src/components/navbar/Navbar.jsx]: give alt

File: src/context/liked-videos/liked-video-reducer.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ export const likedVideoReducer = (state, action) => {
   const { type, payload } = action;
   switch (type) {
     case GET_LIKED_VIDEOS:
-      return [...state, ...payload];
+      return payload.likedVideos;
     case ADD_TO_LIKE:
       console.log(payload);
```
Comment (rkapoor10) [inline src/context/liked-videos/liked-video-reducer.js]: remove console log

File: src/context/videos/VideoContext.jsx [hunk 3]
```diff
@@ -65,9 +66,9 @@ const VideoProvider = ({ children }) => {
           },
         });
     } catch (error) {
-      videoDispatch({type:ERROR, payload:{error:error.message}})
-    }finally{
-      videoDispatch({type:LOADING, payload:{loading:false}})
+      videoDispatch({ type: ERROR, payload: { error: error.message } });
+    } finally {
+      videoDispatch({ type: LOADING, payload: { loading: false } });
```
Comment (rkapoor10) [inline src/context/videos/VideoContext.jsx]: liked this use of finally

File: src/pages/watchlater/WatchLater.jsx [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
+import {useEffect} from "react";
 import VideoCard from "../../components/shared/video-card/VideoCard";
 import { useWatchLaterVideos } from "../../context/watch-later/WatchLaterVideoContext";
 
 const WatchLater = () => {
-  const { watchLaterVideos } = useWatchLaterVideos();
+  const { watchLaterVideos,getWatchLaterVideos  } = useWatchLaterVideos();
+  useEffect(() => {
+    getWatchLaterVideos();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (rkapoor10) [inline src/pages/watchlater/WatchLater.jsx]: why are these comments coming

**Review by rkapoor10 — APPROVED** [review]
the code is well structured and looks good to merge

File: src/pages/watchlater/WatchLater.jsx (hunk 4 above)
Comment (JantuDeb) [reply src/pages/watchlater/WatchLater.jsx]: To disable warnings

**Pull request merged**